pytorch_lightning/plugins/training_type/horovod.py

```diff
@@ -90,12 +90,23 @@ def _filter_named_parameters(model, optimizer):
             return [(name, p) for name, p in model.named_parameters() if p in opt_params]
 
         # Horovod: wrap optimizers to perform gradient aggregation via allreduce
-        optimizers = [
-            hvd.DistributedOptimizer(
-                optimizer, named_parameters=_filter_named_parameters(self.lightning_module, optimizer)
-            ) for optimizer in optimizers
-        ]
-        self.lightning_module.trainer.accelerator.optimizers = optimizers
+        h_optimizers = []
+        for optimizer in optimizers:
+            if 'horovod' not in str(optimizer.__class__):
```

Comment:
```suggestion
            if not isinstance(optimizer, hvd.DistributedOptimizer):
```
isn't it better to check the instance?

Reply:
I've tried this way, but `hvd.DistributedOptimizer ` is a function.

Reply:
oh interesting, do we know what it returns?

Reply:
Below is the similified source code of horovod optimizer wrapper.
```python
class _DistributedOptimizer:
    def __init__(self, params, named_parameters, compression,
                 backward_passes_per_step=1, op=Average,
                 gradient_predivide_factor=1.0,
                 groups=None,
                 sparse_as_dense=False):
        super(self.__class__, self).__init__(params)
        ...


def DistributedOptimizer(optimizer, named_parameters=None,
                         compression=Compression.none,
                         backward_passes_per_step=1,
                         op=Average,
                         gradient_predivide_factor=1.0,
                         num_groups=0, groups=None,
                         sparse_as_dense=False):
        ...
        cls = type(optimizer.__class__.__name__, (optimizer.__class__,),
                   dict(_DistributedOptimizer.__dict__)) 
        return cls(optimizer.param_groups, named_parameters, compression, backward_passes_per_step, op,
                   gradient_predivide_factor, groups, sparse_as_dense)
```

I tried to use `isinstance(optimizer, hvd._DistributedOptimizer)` for the check, but it returns `False`

suppose the original optimizer is `torch.optim.adam.Adam`, then the type of returned object is `horovod.torch.optimizer.Adam`.

Reply:
okay great! 
I think your solution is best then :)